Bono/goth 378

bonomite opened the pull request:
- newsletter & form front end and alt dark version
- hook up newsletter to backend
- NEWSLETTER_API added to nuxt config and circleci files
- created re-usable EmailCollectorForm.vue comp
- updated DS to support dark mode for the checkbox, inputTextField and button-link
- hack for v-card image issue where the image does not load on desktop break point (will investigate at some point)
- more unique footer ID and Class names
- footer responsive breakpoint adjustments

bonomite commented on assets/scss/global.scss:
```diff
@@ -143,3 +143,11 @@ ol ol li {
+
+// to go is DS extentions
+
+// v-card image hack to figure out
```
v-card hack to solve image issue. Will investigate at some point

bonomite commented on nuxt.config.ts:
```diff
@@ -64,6 +64,8 @@ export default defineNuxtConfig({
     API_URL: process.env['API_URL'] || 'https://cms.demo.nypr.digital/api/v2',
     IMAGE_BASE_URL: process.env['IMAGE_BASE_URL'] || 'https://cms.demo.nypr.digital/images/',
     GA_MEASUREMENT_ID: process.env['GA_MEASUREMENT_ID'],
+    NEWSLETTER_API: process.env['NEWSLETTER_API'] || 'https://api.demo.nypr.digital/email-proxy/subscribe',
```
Submitting the form only works with the PROD API

bonomite commented on pages/index.vue:
```diff
@@ -12,6 +11,15 @@ const articles = await findArticlePages('').then(({ data }) =>
 )
 
+const newsletterSubmitEvent = (e) => {
+  //emitted newsletter submit event, @Matt, not exactly sure how to get this work like you mentioned.
```
@walsh9 Was having trouble with the gtag event here... but I emitted it up to the page like you asked.

bonomite commented on plugins/analytics.client.ts:
```diff
@@ -1,28 +1,27 @@
 export default defineNuxtPlugin(() => {
-    const config = useRuntimeConfig()
-    const membershipStatus = useMembershipStatus()
+  const config = useRuntimeConfig()
```
formatting noise... we need to get on the same page with this soon.

walsh9 commented on components/NewsletterHome.vue:
```diff
@@ -0,0 +1,107 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useRuntimeConfig } from '#app'
+import axios from 'axios'
```
It seems like overkill to include all of axios just for the newsletter, I think we cant probably hand the form submission with a regular fetch.

walsh9 commented on components/NewsletterHome.vue:
```diff
@@ -0,0 +1,107 @@
+  submissionStatus.value = null
+  axios
+    .post(config.NEWSLETTER_API, {
+      list: '65dbec786b',
```
This list id shouldn't be hardcoded here, we should put it in a runtime config variable. (If it doesn't change between demo and prod, we don't need to make a whole env var for it, just moving it to the config is fine)

bonomite replied:
done

walsh9 commented on components/EmailCollectorForm.vue:
```diff
@@ -0,0 +1,185 @@
+        >
+        <div class="field-checkbox mt-3">
+          <Checkbox
+            id="binary"
```
We should give this checkbox a better name than binary. what is it for?

bonomite replied:
not needed, I removed the id prop

walsh9 commented on components/GothamistFooter.vue:
```diff
@@ -17,13 +16,17 @@ const copyrightYear = ref(props.navData.copyrightYear)
 <template>
-  <section id="footer" class="footer" data-style-mode="dark">
+  <section
+    id="gotahmist-footer"
```
Looks like a typo here.

bonomite replied:
whoops!

walsh9 commented on components/ScrollToTopButton.vue:
```diff
@@ -22,7 +22,7 @@ const initScrollTrigger = () => {
     gsapScrollTrigger = ScrollTrigger.create({
       id: 'scrollTopID',
-      trigger: '#footer',
+      trigger: '#gotahmist-footer',
```
A matching typo for the earlier one.

walsh9 commented on components/NewsletterHome.vue:
```diff
@@ -0,0 +1,107 @@
+  (e: 'submitEventEmit', value: any): void
+}>()
+
+const showComp = ref(true)
```
Can we spell out the word "Component" here. It's not immediately obvious if this is referring to component or composition or completion or something else here.

walsh9 commented on components/NewsletterHome.vue:
```diff
@@ -0,0 +1,107 @@
+
+const showComp = ref(true)
+const submissionStatus = ref(null)
+const isSubmiting = ref(false)
```
another typo here, should be submitting with two t's

walsh9 reviewed (changes requested):
Looks good overall but I did leave a lot of feedback about names. It might seem like nitpicking but we're gonna have to read this stuff later so I think it's important.

bonomite commented on components/EmailCollectorForm.vue:
```diff
@@ -0,0 +1,185 @@
+const emit = defineEmits<{
+  (e: 'submitFormEmit', value: any): void
+  (e: 'noThanksEventEmit'): void
+}>()
```
ok... I understand. Cool. I guess there is no need to mention it is an emit... because it's clearly an emit... and as a prop, it's preceded with a "@" which clear indicates its is an emit.

bonomite commented:
Great stuff @walsh9 !! Appreciate looking through. I made updates and covered all your comments except the Axios replacement. 
I also realized that I did not add a email format validation method... so I added that to the EmailCollectorForm.vue file. and also changed the logic on how the error text populates. I'll see if I can get the Fetch() in there Monday morning. Have a great long weekend.

walsh9 commented on components/NewsletterHome.vue:
```diff
@@ -41,17 +40,20 @@ const hideComp = () => {
-      },
-    })
+    }),
+  })
```
I think if you use nuxt's $fetch you can shorten this to just something like
```  $fetch(config.NEWSLETTER_API, {
      method: 'POST',
      body: {
        list: config.NEWSLETTER_LIST_ID,
        email: email,
      }
  })
```
https://v3.nuxtjs.org/api/utils/$fetch
https://github.com/unjs/ohmyfetch#%EF%B8%8F-json-body

bonomite replied:
if I remove the "JSON.stringify()" I get this error in VSCODE
![image](https://user-images.githubusercontent.com/13803343/181076415-dbdba60c-d003-4e50-9f08-9fe304f58b70.png)

bonomite replied:
but it seems to submit fine....

bonomite replied:
ah... ok added the "$" to the fetch

walsh9 reviewed (changes requested):
Looks great. Just one last thing
